pkg/kubelet/dockertools/docker.go

```diff
@@ -1049,3 +1049,21 @@ func RunContainer(client DockerInterface, container *api.Container, pod *api.Pod
 	}
 	return dockerContainer.ID, nil
 }
+
+// KillContainer kills a docker container with the given containerID.
+// TODO(yifan): To use strong type for the containerID.
+func KillContainer(client DockerInterface, containerID string,
```

Comment:
Let me just hold this, and take a time to rethink. I will try for the `preparePod`, see how that looks :)